### PR TITLE
[Feature] Eheza Reports

### PR DESCRIPTION
**Pull request by anvmn**
#906 

### Deployment instructions:

```
drush updb
drush fra -y

- Disable AQ task on Jenkins.
- Execute `drush scr profiles/hedley/modules/custom/hedley_reports/scripts/generate-data-for-all.php ` - expected run time ~10 hours.
- Create Jenkins task to execute `drush scr profiles/hedley/modules/custom/hedley_reports/scripts/recalculate-large-datasets.php` once a day during the night.
- Run created task once, right away.
- Create Jenkins task to execute `drush scr profiles/hedley/modules/custom/hedley_ncda/scripts/recalculate-large-datasets.php` once a day during the night.
- Run created task once, right away.
- Enable AQ task on Jenkins.

Only on test and live:
 - drush scr profiles/hedley/modules/custom/hedley_person/scripts/update-creation-date.php
 - drush scr  profiles/hedley/modules/custom/hedley_person/scripts/fix-geofields.php
 ```

### CODE REVIEW

**Comment by anvmn** [comment]
Selection_549

**Comment by anvmn** [comment]
Peek 2024-05-22 15-17

File: server/elm/elm-watch.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ function run {
   date -R;
   echo -en "${COLOR_OFF}";
 
-  elm make src/Main.elm --output ../hedley/modules/custom/hedley_ncda/js/elm-main.js
+  elm make src/Main.elm --output ../hedley/modules/custom/hedley_general/js/elm-main.js
```
Comment (AronNovak) [inline server/elm/elm-watch.sh:23]: Don't we need both?

File: server/elm/src/Translate.elm [hunk 2]
```diff
@@ -384,6 +395,26 @@ translationSet transId =
...
+
+                SelectionOptionDemographics ->
+                    { english = "Demographics"
+                    , kinyarwanda = Nothing
+                    , kirundi = Nothing
+                    }
+
+                SelectionOptionHealthCenter ->
+                    { english = "Health Center"
+                    , kinyarwanda = Nothing
```
Comment (AronNovak) [inline server/elm/src/Translate.elm:838]:
```suggestion
, kinyarwanda = Just "Ikigo Nderabuzima"
```

File: server/elm/src/Translate.elm [hunk 3]
```diff
@@ -384,6 +395,26 @@ translationSet transId =
...
+                    , kirundi = Nothing
```
Comment (AronNovak) [inline server/elm/src/Translate.elm:839]:
```suggestion
                    , kirundi = Just "Ivuriro"
```

File: server/elm/src/Translate.elm (hunk 2 above)
Comment (AronNovak) [reply server/elm/src/Translate.elm:838]: My point is that from the client, we are able to populate translations partially.

File: server/hedley/modules/custom/hedley_reports/hedley_reports.module [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+function hedley_reports_menu() {
+
+  $items['admin/reports/aggregated-reports'] = array(
+    'title' => 'Aggregated Reports',
+    'description' => 'View Aggregated Reports',
+    'page callback' => 'hedley_reports_aggregated_callback_menu',
+    'access callback' => 'hedley_reports_aggregated_reports_report_access',
+  );
+
+//  $items['admin/reports/aggregated-reports/%/%'] = array(
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_reports/hedley_reports.module]: Any reason to not delete commented out lines in this file?

File: server/elm/elm-watch.sh (hunk 1 above)
Comment (anvmn) [reply server/elm/elm-watch.sh:23]: No. Both NCDA and Reports running under common Elm application (different pages).

File: server/elm/src/Translate.elm (hunk 2 above)
Comment (anvmn) [reply server/elm/src/Translate.elm:838]: Well, right now, there's no requirement to have the APP translated, so I don't see a reason to spend time on it.
If translation requirement will come, we'll probably outsource the task to someone for TIP team, as we usually do.

File: server/hedley/modules/custom/hedley_reports/hedley_reports.module (hunk 4 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_reports/hedley_reports.module]: These are used for the Reports page in a next PR...

**Review by AronNovak — APPROVED** [review]

**Pull request merged**